### ISSUE TITLE
org-roam buffer is empty

### ISSUE THREAD

**Issue by zaks** [org-roam/org-roam#998]
#### Steps to Reproduce

Roam config:

  (require 'org-roam)
  (setq org-roam-directory "/Users/zaks/roam")
  (add-hook 'after-init-hook 'org-roam-mode)

#### Backtrace

There is no backtrace after toggling debug on error.
The database contains the links.
There is nothing in the message buffer.

#### Expected Results

The *org-roam* buffer after Emacs restart stopped updating altogether.
org-roam-db-clean followed by org-roam-db-build-cache did not work.

#### Actual Results

Empty buffer.

### Environment

Copy info below this line into issue:
- Emacs: GNU Emacs 26.3 (build 1, x86_64-apple-darwin18.2.0, NS appkit-1671.20 Version 10.14.3 (Build 18D109))
 of 2019-09-02
- Framework: Spacemacs
- Org: Org mode version 9.3.7 (9.3.7-16-g521d7f-elpa @ /Users/zaks/.emacs.d/elpa/26.3/develop/org-20200803/)
- Org-roam: 1.2.1

[current MELPA version]

**Comment by jethrokuan** [comment]
Are you on a specific file? The buffer doesn't show anything if you haven't opened an org-roam file.

**Comment by kisaragi-hiu** [comment]
How does your `org-roam-buffer-prepare-hook` look like? If some part of your configuration accidentally removed its content this might end up happening.

**Comment by zaks** [comment]
Sorry for not replying timely folks - gotta set my github notification settings right.

@jethrokuan - The buffer is empty while I visit an org-roam file - yes. 
@kisaragi-hiu - The value of the hook:

(org-roam-buffer--insert-title org-roam-buffer--insert-backlinks org-roam-buffer--insert-ref-links)

**Comment by zaks** [comment]
I found the cause.I have been configuring the after-init-hook in the user-config section. This did not run at all. 

I had open the org-roam buffers and switched to them. Then I ran the org-roam-mode manuall - but it was too late for the already open files. I have moved the hook setup to the spacemacs/user-init section and all runs fine!

Thanks for looking into the issue

**Comment by bjc** [comment]
I'm having the same issue, but without the same solution.

I've verified that `org-roam-mode` is enabled on startup (via `after-init-hook`), and that it's enabled before I try to open any org roam files via `org-roam-find-file`.

I've run `org-roam-db-clear` and `org-roam-db-update`. While `…clear` doesn't provide any output, I do get a message from `…update` saying that all of the files in my `org-roam-directory` have been modified after it processes them:

`(org-roam) total: Δ21, files-modified: Δ21, ids: Δ0, links: Δ23, tags: Δ19, titles: Δ23, refs: Δ0, deleted: Δ0`

I've even gone so far as to delete the database itself (from `org-roam-db-location`) from my filesystem outside of emacs, restarted emacs, and run the database update command. I cannot figure out how to get the backlinks buffer back, nor even where to begin in trying to debug the issue -- I briefly attempted to use `org-roam-dev-mode`, but it didn't offer any insight.

It is possible that this started when I renamed the autogenerated index file (from `org-roam-jump-to-index`) from `YYYYMMDDHHmmss-index.org` to a plain `index.org` outside of `org-roam`. Unfortunately, I also restarted emacs at the time, so it may also be an issue with some important state not being saved. I'd also not quit emacs after installing and using `org-roam` until then.

Vanilla Emacs version 27.1
Org-Roam version 20210128.1341 from melpa, commit b0fd12647b94ba6e3cf82a2a5b1ee7655ac07760

`org-roam-buffer-prepare-hook` is `(org-roam-buffer--insert-title org-roam-buffer--insert-backlinks org-roam-buffer--insert-ref-links)`

**Comment by kisaragi-hiu** [comment]
"files-modified" is the number of files that has been motified since the last time the cache was built, not files that org-roam modified.

`--clear` basically has the same effect as deleting the cache database.

You can start trying to debug it by looking into the `org-roam-buffer-toggle-display` function itself. (org-roam-dev-mode simply "ensures consistent application of various developer settings." (org-roam-dev.el))

**Comment by bjc** [comment]
I have found the issue, and it is two-fold:

1. I accidentally pasted the output of the `original value` section of `describe-variable` for `org-roam-buffer-prepare-hook`, and
2. I added `org-roam-buffer-activate` to `org-roam-buffer-prepare-hook`. I'm not sure where I picked up to do that -- probably some random outdated documentation I found through late night googing. I'm also kind of amazed the whole thing didn't just crash.

So it's all working now. Thanks!
